test/WebJobs.Script.Tests.Integration/TestScripts/CSharp/Scenarios/run.csx

```diff
@@ -36,6 +36,21 @@ public static void Run(ScenarioInput input, out string blob, TraceWriter trace,
             ["MyCustomMetricProperty"] = 100
         });
     }
+    else if (input.Scenario == "beginScope")
+    {
+        using(logger.BeginScope("{customKey1}", "customValue1"))
```

Comment:
Note: this is what was broken. `ILogger` turns this into an `IEnumerable<KeyValuePair<string, object>>`, which we didn't support. Now we'll turn it into a dictinoary and store the scope, applying the key/value pairs to any logs inside the `using`. Same behavior happens in `WebJobs`, but the `DictionaryLoggerScope` changes needed to be moved here as well since I didn't make that class public.